### PR TITLE
Linkedin/get test addons

### PR DESCRIPTION
**Pull request by amarathavale**
Adding the following changes
1. Batch the data load in batch sizes of 200K in each iteration. Loading 1M records into a collection with 100K RUs took <5 minutes. This is single-threaded, but the degree of parallelism can be increased.
2. Dockerfile: See the top of the docker file for the build step.
    It was executed using: docker run --env-file arguments.list linkedin-ctl-workload, with the following params
ctl_endpoint="value"
ctl_key="value"
ctl_database=CTL
ctl_collection=ctlWorkloadInvitations
ctl_throughput=100000
ctl_concurrency=25
ctl_number_of_precreated_documents=1000000
ctl_number_of_operations=-1
ctl_max_running_time_duration=PT15m
ctl_printing_interval=60

The latency metrics were printed to the console every 60s.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-benchmark/src/test/java/com/azure/cosmos/benchmark/linkedin/data/TestInvitationDataGenerator.java
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/test/java/com/azure/cosmos/benchmark/linkedin/data/TestInvitationDataGenerator.java:14]: as code style we don't add "test" prefix to the our test methods.

**Review by moderakh — APPROVED** [review]
LGTM.

Please wait for @simplynaveen20 review and sign off.

File: sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/Dockerfile
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/Dockerfile:5]: Why we need separate docker image , is it for linux/java version ?  @kushagraThapar  fyi, if need separate docker image , then we have to make change in our ctl image uploader yml, to include this image as well

File: sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh:75]: I notice this is 10 min and got copied from sdk run file, however in sdk we give explicit option ctl_max_running_time_duration="PT24H" on ctl run, so either make sure we pass this in LI ctl as well , or make this default as PT24H here itself.

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/DataLoader.java
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/DataLoader.java:100]: Nice , thanks for introducing fault tolerance.

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/GetTestRunner.java
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/GetTestRunner.java:87]: What is the motivation here ? We should not stop the run on fixed error count, instead let the run complete and analysis the result later

File: sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh:61]: Can we make default to 1M , that's what we wanted for LI workload right  ?

File: sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/Dockerfile
Comment (amarathavale) [reply sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/Dockerfile:5]: We wanted to come as close as possible to the conditions under which our application runs, the OS, java sdk, the JVM options (which haven't been configured yet). That's the reason for the lentos and Zulu jvm steps.

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/GetTestRunner.java
Comment (amarathavale) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/linkedin/GetTestRunner.java:87]: After a few runs, I realize it's not adding much value. The error stats are counting and reported separately, so removing it.

File: sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh
Comment (amarathavale) [reply sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh:61]: For our workload we want it ~10M+. From our email thread with Shelly, we should be able to allocate a collection with 1M RUs, and load unto 5TB of data. We will have to experiment with loading a larger and larger set of documents, and see how it impacts the data load times

Changed the default value to 1M here.

File: sdk/cosmos/azure-cosmos-benchmark/src/test/java/com/azure/cosmos/benchmark/linkedin/data/TestInvitationDataGenerator.java
Comment (amarathavale) [reply sdk/cosmos/azure-cosmos-benchmark/src/test/java/com/azure/cosmos/benchmark/linkedin/data/TestInvitationDataGenerator.java:14]: Fixed it in the latest revision

**Comment by amarathavale** [comment]
Had to edit the metric prefix/name to make it work with the graphite dashboard. See the render below for the plot. Metrics were sent previously too, as was evident in the logs

[aathaval@aathaval-mn2 cosmos]$ docker logs 9d1d98328144 | less
....
08/02/2021 15:39:35 :: [tagdb] Tagging LinkedInCtlWorkload.GET-ctlWorkloadInvitations-Latency.p99
08/02/2021 15:39:35 :: [tagdb] Tagged LinkedInCtlWorkload.GET-ctlWorkloadInvitations-Latency.p99 in 0.004624128341674805

Screen Shot 2021-02-08 at 7 41 45 AM

File: sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh:26]: Please change default name with li prefix here and in collection name.

**Review by simplynaveen20 — APPROVED** [review]
LGTM , please address comment for db and collection name. After that we are good to merge this. Thanks Amar

File: sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-benchmark/ctl/linkedin/run_benchmark.sh:26]: is  this resolved? I am unable to see the new changes

File: sdk/cosmos/azure-cosmos-benchmark/ctl/run_benchmark.sh
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/ctl/run_benchmark.sh:75]: Please revert this is SDK file

File: sdk/cosmos/azure-cosmos-benchmark/ctl/run_benchmark.sh
Comment (simplynaveen20) [inline sdk/cosmos/azure-cosmos-benchmark/ctl/run_benchmark.sh:88]: This change wont have any effect , still we should keep the file intact if we not meant to change any thing.

**Pull request merged**
